From 1bfbb066794249725133b7b42d767dc9013a6839 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Wed, 6 Feb 2013 10:44:11 +0100
Subject: [PATCH] xilinx: dts: Update zedboard dts

Generated by the latest device-tree BSP with one exception
which is manually adding flash node.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 arch/arm/boot/dts/zynq-zed.dts |   74 ++++++++++++++++++---------------------
 1 files changed, 34 insertions(+), 40 deletions(-)

diff --git a/arch/arm/boot/dts/zynq-zed.dts b/arch/arm/boot/dts/zynq-zed.dts
index 897fa49..390c4bc 100644
--- a/arch/arm/boot/dts/zynq-zed.dts
+++ b/arch/arm/boot/dts/zynq-zed.dts
@@ -23,8 +23,8 @@
  * MA 02111-1307 USA
  *
  * CAUTION: This file is automatically generated by libgen.
- * Version: Xilinx EDK 14.3 EDK_P.40xd
- * Today is: Tuesday, the 27 of November, 2012; 12:29:21
+ * Version: Xilinx EDK 14.4 EDK_P.49d
+ * Today is: Wednesday, the 06 of February, 2013; 10:43:23
  *
  * XPS project directory: device-tree_bsp_0
  */
@@ -48,7 +48,6 @@
 		#cpus = <0x2>;
 		#size-cells = <0>;
 		ps7_cortexa9_0: cpu@0 {
-			clock-frequency = <666666687>;
 			compatible = "xlnx,ps7-cortexa9-1.00.a";
 			d-cache-line-size = <0x20>;
 			d-cache-size = <0x8000>;
@@ -57,12 +56,10 @@
 			i-cache-size = <0x8000>;
 			model = "ps7_cortexa9,1.00.a";
 			reg = <0>;
-			timebase-frequency = <333333343>;
 			xlnx,cpu-1x-clk-freq-hz = <0x69f6bcb>;
 			xlnx,cpu-clk-freq-hz = <0x27bc86bf>;
 		} ;
 		ps7_cortexa9_1: cpu@1 {
-			clock-frequency = <666666687>;
 			compatible = "xlnx,ps7-cortexa9-1.00.a";
 			d-cache-line-size = <0x20>;
 			d-cache-size = <0x8000>;
@@ -71,11 +68,16 @@
 			i-cache-size = <0x8000>;
 			model = "ps7_cortexa9,1.00.a";
 			reg = <1>;
-			timebase-frequency = <333333343>;
 			xlnx,cpu-1x-clk-freq-hz = <0x69f6bcb>;
 			xlnx,cpu-clk-freq-hz = <0x27bc86bf>;
 		} ;
 	} ;
+	pmu {
+		compatible = "arm,cortex-a9-pmu";
+		interrupt-parent = <&ps7_scugic_0>;
+		interrupts = < 0 5 4 0 6 4 >;
+		reg = < 0xf8891000 0x1000 0xf8893000 0x1000 >;
+	} ;
 	ps7_ddr_0: memory@0 {
 		device_type = "memory";
 		reg = < 0x0 0x20000000 >;
@@ -109,15 +111,13 @@
 		ps7_dev_cfg_0: ps7-dev-cfg@f8007000 {
 			compatible = "xlnx,ps7-dev-cfg-1.00.a";
 			interrupt-parent = <&ps7_scugic_0>;
-			interrupts = < 0 8 0 >;
+			interrupts = < 0 8 4 >;
 			reg = < 0xf8007000 0x1000 >;
 		} ;
-		ps7_dma_ns: ps7-dma@f8004000 {
-			compatible = "xlnx,ps7-dma-1.00.a";
-			reg = < 0xf8004000 0x1000 >;
-		} ;
 		ps7_dma_s: ps7-dma@f8003000 {
-			compatible = "xlnx,ps7-dma-1.00.a";
+			compatible = "xlnx,ps7-dma-1.00.a", "arm,pl330";
+			interrupt-parent = <&ps7_scugic_0>;
+			interrupts = < 0 13 4 0 14 4 0 15 4 0 16 4 0 17 4 0 40 4 0 41 4 0 42 4 0 43 4 >;
 			reg = < 0xf8003000 0x1000 >;
 		} ;
 		ps7_ethernet_0: ps7-ethernet@e000b000 {
@@ -125,7 +125,7 @@
 			#size-cells = <0>;
 			compatible = "xlnx,ps7-ethernet-1.00.a";
 			interrupt-parent = <&ps7_scugic_0>;
-			interrupts = < 0 22 0 >;
+			interrupts = < 0 22 1 >;
 			local-mac-address = [ 00 0a 35 00 00 00 ];
 			phy-handle = <&phy0>;
 			reg = < 0xe000b000 0x1000 >;
@@ -158,7 +158,7 @@
 			gpio-mask-high = <0xc0000>;
 			gpio-mask-low = <0xfe81>;
 			interrupt-parent = <&ps7_scugic_0>;
-			interrupts = < 0 20 0 >;
+			interrupts = < 0 20 4 >;
 			reg = < 0xe000a000 0x1000 >;
 		} ;
 		ps7_iop_bus_config_0: ps7-iop-bus-config@e0200000 {
@@ -171,18 +171,15 @@
 			cache-level = < 2 >;
 			cache-unified ;
 			compatible = "xlnx,ps7-pl310-1.00.a", "arm,pl310-cache";
+			interrupt-parent = <&ps7_scugic_0>;
+			interrupts = < 0 2 4 >;
 			reg = < 0xf8f02000 0x1000 >;
 		} ;
-		pmu {
-			compatible = "arm,cortex-a9-pmu";
-			interrupts = <0 5 4>, <0 6 4>;
-			interrupt-parent = <&ps7_scugic_0>;
-		};
 		ps7_qspi_0: ps7-qspi@e000d000 {
 			bus-num = <0>;
 			compatible = "xlnx,ps7-qspi-1.00.a";
 			interrupt-parent = <&ps7_scugic_0>;
-			interrupts = < 0 19 0 >;
+			interrupts = < 0 19 4 >;
 			is-dual = <0>;
 			num-chip-select = <1>;
 			reg = < 0xe000d000 0x1000 >;
@@ -225,14 +222,6 @@
 			reg = < 0xfc000000 0x1000000 >;
 			xlnx,qspi-clk-freq-hz = <0xe4e1c0>;
 		} ;
-		ps7_ram_0: ps7-ram@0 {
-			compatible = "xlnx,ps7-ram-1.00.a";
-			reg = < 0x0 0x30000 >;
-		} ;
-		ps7_ram_1: ps7-ram@ffff0000 {
-			compatible = "xlnx,ps7-ram-1.00.a";
-			reg = < 0xffff0000 0xfe00 >;
-		} ;
 		ps7_scugic_0: ps7-scugic@f8f01000 {
 			#address-cells = < 2 >;
 			#interrupt-cells = < 3 >;
@@ -244,25 +233,22 @@
 			reg = < 0xf8f01000 0x1000 0xf8f00100 0x100 >;
 		} ;
 		ps7_scutimer_0: ps7-scutimer@f8f00600 {
-			clock-frequency = <333333343>;
 			compatible = "xlnx,ps7-scutimer-1.00.a", "arm,cortex-a9-twd-timer";
 			interrupt-parent = <&ps7_scugic_0>;
-			interrupts = < 1 13 0 >;
+			interrupts = < 1 13 769 >;
 			reg = < 0xf8f00600 0x20 >;
 		} ;
 		ps7_scuwdt_0: ps7-scuwdt@f8f00620 {
-			clock-frequency = <333333343>;
 			compatible = "xlnx,ps7-scuwdt-1.00.a";
 			device_type = "watchdog";
 			interrupt-parent = <&ps7_scugic_0>;
-			interrupts = < 1 14 0 >;
+			interrupts = < 1 14 769 >;
 			reg = < 0xf8f00620 0xe0 >;
 		} ;
 		ps7_sd_0: ps7-sdio@e0100000 {
-			clock-frequency = <50000000>;
 			compatible = "xlnx,ps7-sdio-1.00.a", "generic-sdhci";
 			interrupt-parent = <&ps7_scugic_0>;
-			interrupts = < 0 24 0 >;
+			interrupts = < 0 24 4 >;
 			reg = < 0xe0100000 0x1000 >;
 			xlnx,has-cd = <0x1>;
 			xlnx,has-power = <0x0>;
@@ -270,23 +256,31 @@
 			xlnx,sdio-clk-freq-hz = <0x2faf080>;
 		} ;
 		ps7_slcr_0: ps7-slcr@f8000000 {
-			compatible = "xlnx,ps7-slcr-1.00.a";
+			compatible = "xlnx,ps7-slcr-1.00.a", "xlnx,zynq-slcr";
 			reg = < 0xf8000000 0x1000 >;
+			clocks {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				ps_clk: ps_clk {
+					#clock-cells = <0>;
+					clock-frequency = <33333333>;
+					clock-output-names = "PS_CLK";
+					compatible = "fixed-clock";
+				} ;
+			} ;
 		} ;
 		ps7_ttc_0: ps7-ttc@f8001000 {
 			compatible = "xlnx,ps7-ttc-1.00.a";
 			interrupt-parent = <&ps7_scugic_0>;
-			interrupts = < 0 10 0 0 11 0 0 12 0 >;
+			interrupts = < 0 10 4 0 11 4 0 12 4 >;
 			reg = < 0xf8001000 0x1000 >;
 		} ;
 		ps7_uart_1: serial@e0001000 {
-			clock = <50000000>;
-			clock-frequency = <50000000>;
 			compatible = "xlnx,ps7-uart-1.00.a", "xlnx,xuartps";
 			current-speed = <115200>;
 			device_type = "serial";
 			interrupt-parent = <&ps7_scugic_0>;
-			interrupts = < 0 50 0 >;
+			interrupts = < 0 50 4 >;
 			port-number = <0>;
 			reg = < 0xe0001000 0x1000 >;
 			xlnx,has-modem = <0x0>;
@@ -296,7 +290,7 @@
 			compatible = "xlnx,ps7-usb-1.00.a";
 			dr_mode = "host";
 			interrupt-parent = <&ps7_scugic_0>;
-			interrupts = < 0 21 0 >;
+			interrupts = < 0 21 4 >;
 			phy_type = "ulpi";
 			reg = < 0xe0002000 0x1000 >;
 			xlnx,usb-reset = <0xffffffff>;
-- 
1.7.1

